### PR TITLE
Add tests for number bodies

### PR DESCRIPTION
**Pull request by pakrym**
Requires new test-server https://github.com/Azure/autorest.testserver/pull/91

Adds to https://github.com/Azure/autorest.csharp/issues/286

### CODE REVIEW

File: test/AutoRest.TestServer.Tests/NumberTest.cs [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+using NUnit.Framework;
+
+namespace AutoRest.TestServer.Tests
+{
+    public class NumberTest : TestServerTestBase
+    {
+        public NumberTest(TestServerVersion version) : base(version, "number") { }
+
+        [Test]
+        public Task putFloatBigScientificNotation() => TestStatus(async (host, pipeline) => await NumberOperations.PutBigFloatAsync(ClientDiagnostics, pipeline, 3.402823e+20f, host));
```
Comment (MiYanni) [inline test/AutoRest.TestServer.Tests/NumberTest.cs]: Do we care if the test names use camelCase versus PascalCase?

File: test/AutoRest.TestServer.Tests/NumberTest.cs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+        [Test]
+        public Task putDoubleBigScientificNotation() => TestStatus(async (host, pipeline) => await NumberOperations.PutBigDoubleAsync(ClientDiagnostics, pipeline, 2.5976931e+101d, host));
+
+        [Test]
+        [IgnoreOnTestServer(TestServerVersion.V2, "Match too strict")]
+        public Task putDecimalBigPositiveDecimal() => TestStatus(async (host, pipeline) => await NumberOperations.PutBigDecimalPositiveDecimalAsync(ClientDiagnostics, pipeline, host));
+
+        [Test]
+        [Ignore("Value 2.5976931e+101 is out of range of C# decimal")]
```
Comment (MiYanni) [inline test/AutoRest.TestServer.Tests/NumberTest.cs]: Hmm. Do you think we need to use a different type than `decimal`? Or, would we just change the tests on the test server?

File: test/AutoRest.TestServer.Tests/NumberTest.cs (hunk 1 above)
Comment (pakrym) [reply test/AutoRest.TestServer.Tests/NumberTest.cs]: Yes, I'll fix

File: test/AutoRest.TestServer.Tests/NumberTest.cs (hunk 2 above)
Comment (pakrym) [reply test/AutoRest.TestServer.Tests/NumberTest.cs]: This is an optional test, I'll file an issue to follow up

Comment (pakrym) [reply test/AutoRest.TestServer.Tests/NumberTest.cs]: https://github.com/Azure/autorest.csharp/issues/335

**Pull request merged**